### PR TITLE
Enhanced search box

### LINKED ISSUE #215: Improve search box
### Describe the feature

I would like to add a clear input button in faq search and also the clear option should appear when the input length > 0.

### Why do you think this feature should be implemented?

_No response_

### Additional context

The clear button should appear only if the input length is > 0
IMG_20231209_135820

A clear input button here
IMG_20231209_135830

Please assign me this.

### Additional Information

- [X] I have used the search function to check if an issue already exists
- [x] I'd like to work on this issue

### PR DESCRIPTION
**Pull request by jamnxdev**
# Improved search box in projects and FAQ page
The clear button will not be visible when query is empty and added a clear button on FAQ page

close #215 

https://github.com/kossiitkgp/KWoC-Frontend/assets/65119631/3cf128b3-abf1-45ab-bcdd-0acf64d6287a

### CODE REVIEW

File: src/pages/FAQ.tsx [hunk 1]
```diff
@@ -30,16 +32,22 @@ export default function FAQ() {
               Wanna ask us a question? Check these first.
             </p>
           </section>
-
           <div className="p-4 my-4 mx-0">
-            <div className="w-full bg-none border-none">
+        <div className="flex py-4 px-6 rounded-md outline-none w-[80vw] max-w-3xl border-none text-white bg-slate-900 font-semibold bg-none">
```
Comment (Nafisay) [inline src/pages/FAQ.tsx]: can you indent this properly (just to look professional)

File: src/pages/Projects.tsx [hunk 2]
```diff
@@ -59,11 +59,14 @@ function Projects() {
...
           ></input>
+          {query !== "" && (
           <button onClick={() => setQuery("")}>
-            <IconContext.Provider value={{ size: "1.8rem" }}>
+            <IconContext.Provider value={{ size: "1.6rem" }}>
               <MdCancel />
             </IconContext.Provider>
           </button>
+          )
+          }
```
Comment (Nafisay) [inline src/pages/Projects.tsx]: put this brace in the above line (to maintain uniformity with the other codes)

File: src/pages/FAQ.tsx [hunk 3]
```diff
@@ -30,16 +32,22 @@ export default function FAQ() {
               Wanna ask us a question? Check these first.
             </p>
           </section>
-
```
Comment (Nafisay) [inline src/pages/FAQ.tsx:33]: please keep this extra line, the code looks clean this way

**Review by Nafisay — CHANGES_REQUESTED** [review]

**Comment by Nafisay** [comment]
Good work!

**Review by Nafisay — APPROVED** [review]

**Review by harshkhandeparkar — APPROVED** [review]

**Comment by harshkhandeparkar** [comment]
Thanks @Jaimin25! :rocket:

**Pull request merged**